package.json

```diff
@@ -51,37 +51,45 @@
 		},
 		"views": {
 			"sourcery-explorer": [
+				{
+					"id": "sourcery.coding_assistant_opt_in",
+					"name": "Coding Assistant",
+					"type": "webview",
+					"icon": "sourcery-icon.png",
+					"contextualTitle": "Coding Assistant",
+					"when": "sourcery.features.coding_assistant && !sourcery.config.coding_assistant.enabled"
+				},
 				{
 					"id": "sourcery.chat",
 					"name": "Coding Assistant",
 					"type": "webview",
 					"icon": "sourcery-icon.png",
 					"contextualTitle": "Coding Assistant",
-					"when": "sourcery.features.coding_assistant"
+					"when": "sourcery.features.coding_assistant && sourcery.config.coding_assistant.enabled === true"
 				},
 				{
 					"id": "sourcery.troubleshooting",
 					"name": "Troubleshooting",
 					"type": "webview",
 					"icon": "sourcery-icon.png",
 					"contextualTitle": "Troubleshooting",
-					"when": "sourcery.features.coding_assistant && sourcery.features.coding_assistant_troubleshooting && sourcery.features.semantic_search"
+					"when": "sourcery.features.coding_assistant && sourcery.config.coding_assistant.enabled === true && sourcery.features.coding_assistant_troubleshooting && sourcery.features.semantic_search"
```

Comment:
**issue:** I think this is a good example of one of the reasons I'm concerned with the approach taken in the core PR - despite the fact that the features in `core` now take configuration, it still has to be explicitly handled here - yuck.